File: code/c/disk_format.c
#include "disk_format.h"
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "utils.h"

void format_disk(const char *disk, const char *fs_choice) {
    char partition[256];
    snprintf(partition, sizeof(partition), "/dev/%s1", disk);

    char command[256];
    if (strcmp(fs_choice, "ntfs") == 0) {
        snprintf(command, sizeof(command), "mkfs.ntfs -f %s", partition);
    } else if (strcmp(fs_choice, "ext4") == 0) {
        snprintf(command, sizeof(command), "mkfs.ext4 %s", partition);
    } else if (strcmp(fs_choice, "vfat") == 0) {
        snprintf(command, sizeof(command), "mkfs.vfat -F 32 %s", partition);
    } else {
        fprintf(stderr, "Unsupported filesystem: %s\n", fs_choice);
        exit(EXIT_FAILURE);
    }

    printf("Formatting %s as %s...\n", partition, fs_choice);
    if (!run_command(command)) {
        fprintf(stderr, "Error formatting partition %s.\n", partition);
        exit(EXIT_FAILURE);
    }

    printf("Partition %s formatted successfully.\n", partition);
}

File: code/python/disk_format.py
import logging
from utils import run_command
import subprocess
import sys

logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")

def format_disk(disk, fs_choice):
    partition = f"/dev/{disk}1"
    
    try:
        if fs_choice == "ntfs":
            logging.info(f"Formatting {partition} as NTFS...")
            run_command(["mkfs.ntfs", "-f", partition])
        elif fs_choice == "ext4":
            logging.info(f"Formatting {partition} as EXT4...")
            run_command(["mkfs.ext4", partition])
        elif fs_choice == "vfat":
            logging.info(f"Formatting {partition} as VFAT...")
            run_command(["mkfs.vfat", "-F", "32", partition])

        logging.info(f"Partition {partition} formatted successfully.")
    except FileNotFoundError:
        logging.error(f"Error: Filesystem utility not found for {fs_choice}. Ensure necessary tools are installed.")
        sys.exit(2)
    except subprocess.CalledProcessError:
        logging.error(f"Error: Failed to format {partition}.")
        sys.exit(1)
